src/graphics/image.c

```diff
@@ -307,22 +312,21 @@ static void draw_compressed_blend_alpha(const image *img, const color_t *data, i
                 x += b;
             } else {
                 data += b;
-                color_t *dst = graphics_get_pixel(x_offset + x, y_offset + y);
                 if (unclipped) {
-                    x += b;
                     while (b) {
-                        color_t d = *dst;
-                        *dst = MIX(color, d, alpha, 0) | MIX(color, d, alpha, 8) | MIX(color, d, alpha, 16);
-                        dst++;
+                        color_t d = dst[x];
```

Comment:
Did you run a profiler to see whether this really makes a difference?

Reply:
I just ran a simple test program to verify this. The old implementation using pointers is 5-10% _faster_ than the new implementation using array indexing, both without (`-O0`) and with (`-O3`) compiler optimizations.